addons/web/static/tests/views/form/form_view_tests.js

```diff
@@ -4021,6 +4021,33 @@ QUnit.module("Views", (hooks) => {
         assert.containsNone(target, ".o_cp_action_menus span:contains(Unarchive)");
     });
 
+    QUnit.test("archive action not shown with readonly active field", async function (assert) {
+
+        // add active field on partner model, but do not put it in the view
+        serverData.models.partner.fields.active = {
+            string: "Active",
+            type: "char",
+            default: true,
+            readonly: true,
+        };
+        await makeView({
+            type: "form",
+            resModel: "partner",
+            resId: 1,
+            serverData,
+            arch: `<form><field name="foo"/></form>`,
```

Comment:
add the field "active"

Reply:
If I add the field "active" in the view, should I leave it in the partner model ?

Reply:
Why don't you try ?

Reply:
I think it's ok now.